src/System.Net.Primitives/tests/PalTests/SocketAddressPalTest.cs

```diff
@@ -61,9 +61,19 @@ public void Port_Get_Set_IPv6(ushort port)
         }
 
         [Fact]
-        public void Port_Get_Set_Throws()
+        public void Port_Get_Set_Throws_IPv4()
```

Comment:
@pgavlin  @stephentoub We need to be able to filter out tests as machines may or may not support all protocols. 
In our in-progress branch, I've started adding `[Trait("IPv4", "true")]` and IPv6, allowing filtering based on what the local machine supports.

I am also working on a `Capabilities` class that will be required for all tests that are making assumptions about the local machine. E.g.:
`Assert.True(Capability.IPv4Support());`

If the Trait filtering works with Theory tests (and can separate the two variations into two traits) then I'm fine with the proposed solution. 

For simplicity, I prefer having 2 separate tests, even if the code seems duplicated.